mlir/lib/Dialect/Tosa/Transforms/TosaFolders.cpp

```diff
@@ -1332,6 +1332,120 @@ struct TosaFoldConstantMaximum
   }
 };
 
+template <typename AccumulatorType, typename InputType>
+SmallVector<AccumulatorType> matmul(
+    ShapedType outputType, ElementsAttr matrixA, ElementsAttr matrixB,
+    const std::function<AccumulatorType(const InputType &)> &convertToAccType) {
```

Comment:
Turn this into a template argument 
```suggestion
    ConvertToAccType &&convertToAccType) {
```
to allow the function to be inlined.